src/lib.rs

```diff
@@ -821,14 +820,13 @@ impl TsunamiBuilder {
             Either::A(future)
         } else {
             Either::B(wait_for_instances)
-        }.and_then(|wait_result| {
-            if let WaitForInstancesResult::Ok(instances, id_to_name) = wait_result {
+        }.and_then(|wait_result| match wait_result {
+            RunningInstanceSet::All(instances, id_to_name) => {
                 Either::A(futures::future::ok((id_to_name, instances)))
-            } else {
-                Either::B(futures::future::err(format_err!(
-                    "not all instances were launched within the time limit"
-                )))
             }
+            RunningInstanceSet::Only(_) => Either::B(futures::future::err(format_err!(
```

Comment:
Yeah, notice how the instance list is dropped here. It needs to somehow be communicated to the cleanup code so that any instances there will be terminated!